docs/reference/reference-training/training/api-pytorch-trainer-reference.rst

```diff
@@ -0,0 +1,120 @@
+###############################
+ PyTorch Trainer API Reference
+###############################
+
++----------------------------------------------------+
+| User Guide                                         |
++====================================================+
+| :doc:`/training/apis-howto/api-pytorch-trainer-ug` |
++----------------------------------------------------+
+
+.. code::
+
+   determined.pytorch.init(
+       *,
+       hparams: Optional[Dict] = None,
+       exp_conf: Optional[Dict[str, Any]] = None,
+       distributed: Optional[core.DistributedContext] = None,
+       aggregation_frequency: int = 1,
+   ) -> pytorch.PyTorchTrialContext:
+
+``pytorch.init()`` builds a ``pytorch.PyTorchTrialContext`` for use with ``PyTorchTrial``.
+
+Always use this method to construct a ``PyTorchTrialContext`` instead of instantiating the class
+directly.
+
+All of the arguments are optional, but ``hparams`` and ``exp_conf`` are used to set the
+corresponding variables on ``PyTorchTrialContext``. So if not passed in, calling
+``context.get_hparams()`` or ``context.get_experiment_config()`` will fail in local-training mode.
+
+``aggregation_frequency`` configures the number of batches before gradients are exchanged in
+   distributed training. This value is configured here because it is used in context.wrap_optimizer.
+
+``DistributedContext`` can be optionally passed in to manually configure distributed training;
+otherwise, it will be automatically initialized from the launch layer.
+
+All ``trainer.*`` calls must be within the scope of this ``with pytorch.init() as trial_context``,
+as there are resources necessary for training which start in the **enter** method and must be
+cleaned up in the corresponding **exit**\ () method.
+
+********************************
+ ``determined.pytorch.Trainer``
+********************************
+
+``class determined.pytorch.Trainer(trial: pytorch.PyTorchTrial, context:
+pytorch.PyTorchTrialContext)``
+
+``Trainer`` is the main class for Trainer API. It has the following required arguments: - ``trial``:
+an instance of the ``PyTorchTrial`` class - ``context``: the ``PyTorchTrialContext`` returned from
+``pytorch.init()``
```

Comment:
probably this should just be done via autodocs, but in case we decide not to, this bulleted list was probably indented wrong and `rstfmt` got confused and joined all the lines together.